File: Retos/Reto #8 - EL GENERADOR PSEUDOALEATORIO [Media]/javascript/jaimegamm.js
/*
 * Crea un generador de números pseudoaleatorios entre 0 y 100.
 * - No puedes usar ninguna función "random" (o semejante) del lenguaje de programación seleccionado.
 *
 * Es más complicado de lo que parece...
 */

// Inicialización de variables
let seed = 42;  // Puedes cambiar el valor inicial (semilla)
const a = 1664525;
const c = 1013904223;
const m = 2**32;

// Función para generar números pseudoaleatorios entre 0 y 100
function generarPseudoAleatorio() {
    seed = (a * seed + c) % m;
    return (seed % 101);  // Módulo 101 para obtener valores entre 0 y 100
}

// Ejemplo de uso
for (let i = 0; i < 10; i++) {
    let numeroAleatorio = generarPseudoAleatorio();
    console.log(numeroAleatorio);
}

File: Retos/Reto #8 - EL GENERADOR PSEUDOALEATORIO [Media]/python/jaimegamm.py
#
#Crea un generador de números pseudoaleatorios entre 0 y 100.
#No puedes usar ninguna función "random" (o semejante) del lenguaje de programación seleccionado.
#
#

class PseudoRandomGenerator:
    def __init__(self, seed):
        self.seed = seed

    def random(self):
        self.seed = (self.seed * 1103515245 + 12345) & 0x7fffffff
        return (self.seed // 65536) % 100

# Ejemplo de uso
seed = 42  # Cambia la semilla según desees
prng = PseudoRandomGenerator(seed)

for _ in range(10):
    print(prng.random())
